packages/internals/config/src/config.test.js

```diff
@@ -0,0 +1,222 @@
+      config.addAccessor('foo.bar', accessorSpy);
+      config.get('foo.bar');
+      expect(accessorSpy.calledWith('baz', config)).to.equal(true);
+    });
+  });
+
+  describe('.set()', function () {
+    it('sets a config property via dot-notation syntax', function () {
+      const config = new Config({});
+      config.set('foo.bar', 'baz');
+      expect(config.get('foo.bar')).to.equal('baz');
+    });
+
+    it('returns the config class instance', function () {
+      const config = new Config({});
+      expect(config.set('bar', 'foo')).to.equal(config);
+    });
+
+    it('throws an error if setting a value invalidates the config schema', function () {
+      const config = new Config({schema});
+      expect(() => config.set('foo', 123)).to.not.throw();
+      expect(() => config.set('foo', 'bar')).to.throw();
+    });
+  });
+
+  describe('.getData()', function () {
+    it('retrieves a config property via dot-notation syntax', function () {
+      const data = {
+        one: 'two',
+        three: {
+          nested: 'four'
+        }
+      };
+      const config = new Config({data});
+      expect(config.getData('three.nested')).to.eql(data.three.nested);
+    });
+    it('does not run the result through accessors', function () {
+      const config = new Config({
+        data: {
+          foo: {
+            bar: 'baz'
+          }
+        }
+      });
+      config.addAccessor('foo.bar', value => '!' + value);
+      config.addAccessor('foo', value => '@' + value);
+      expect(config.getData('foo.bar')).to.equal('baz');
+    });
+  });
+
+  describe('.addAccessor()', function () {
+    it('adds an accessor', function () {
+      const config = new Config();
+      config.addAccessor('foo.bar', val => val);
+      const accessor = config.accessors.find(acc => acc.path === 'foo.bar');
+      expect(accessor).to.be.an('object');
+      expect(accessor).to.have.property('path');
+      expect(accessor).to.have.property('handler');
+    });
+
+    it('attempts to require a bundled accessor by name if the handler argument is a string', function () {
+      const packageLoader = require('./accessors/package-loader');
+      const config = new Config();
+      config.addAccessor('foo.bar', 'package-loader');
+      const accessor = config.accessors.find(acc => acc.path === 'foo.bar');
+      expect(accessor.handler).to.equal(packageLoader);
+    });
+  });
+
+  describe('.getAccessorsForPath()', function () {
+    it('retrieves all accessors that match the dot-notation path', function () {
+      const config = new Config();
+      config.addAccessor('foo.bar', val => val);
+      config.addAccessor('foo.bar.baz', val => val);
+      config.addAccessor('bax.foo.bar', val => val);
+      config.addAccessor('fooooo', val => val);
+      config.addAccessor('bar', val => val);
+      expect(config.getAccessorsForPath('foo.bar').length).to.equal(1);
+      expect(config.getAccessorsForPath('foo.bar.baz').length).to.equal(2);
```

Comment:
Why is this '2' - is it because of the preceding accessor? If so, not sure I would've expected that.